### PR TITLE
Reencode truth and predictions to match labels lexicographic order

### PR DESCRIPTION
**Pull request by sebhrusen**
When framework was providing the combination:
```python
prediction_labels=["b", "a"]
target_encoded=True
```
with therefore "b" encoded as 0, "a" encoded as 1.

Then the `amlb` app was reordering prediction columns to get ["a", "b"], but it was not reencoding predictions and truth values, so that 1 was then meaning "b" instead of "a" after the reordering.

This PR fixes this issues, and ensures that values are re-encoded if needed.

### CODE REVIEW

**Comment by sebhrusen** [comment]
discovered when running `validation` benchmarks on https://github.com/openml/automlbenchmark/pull/153

This was clearly visible on tasks like `APSFailure` (AUC < 0.5), or `micro-mass` (highest probability was not matching prediction).
And then also visible on fold 1 for `iris` on `test` benchmark.

Will create an Issue so that we have a proper test or a validation option that would verify that the `predictions.csv` are correct, which means for example that for classification, we should always have 
prediction value = column with highest probability.

**Comment by PGijsbers** [comment]
Why was it that in the original paper the reported AUC for APSFailure was `>0.5`? ([figure](https://openml.github.io/automlbenchmark/results.html))

**Comment by sebhrusen** [comment]
@PGijsbers, just updated my comment: this was occuring only for MLPlan framework, this is because MLPlan was using a combination of parameters (the ones in this PR description) that was not used by other frameworks until now.

File: docs/HOWTO.md [hunk 1]
```diff
@@ -909,13 +909,27 @@ _Examples of method duration info when using this custom profiling_:
...
+The setup is created for Debian-based linux environments, and macOS (most frameworks can be installed on macOS, ideally with `brew` installed, but there may be a few exceptions), so it may work with other Linux environments not listed above (e.g. Debian, Ubuntu 20.04, ...).
+The best way to run benchmarks on non-supported OS, is to use the docker mode.
+
+If the setup fails on a supported environment, please try the following:
+- force the setup: see above.
+- ensure that the same framework is not set up multiple times in parallel on the same machine:
+  - first use `python runbenchmark.py MyFramework -s only` on one terminal.
+  - then you can trigger multiple `python runbenchmark.py MyFramework ...` (without `-s` option) in parallel.
+- delete the `lib` and `venv` folders, if present, under the given framework folder (e.g. `frameworks/MyFramework`), and try the setup again.
+
```
Comment (sebhrusen) [inline docs/HOWTO.md:931]: not related with this PR, but I was discussing https://github.com/openml/automlbenchmark/issues/157 at the same time, to including this here.

**Review by PGijsbers — APPROVED** [review]

**Pull request merged**